tests/test_s3_client.py

```diff
@@ -73,3 +73,25 @@ async def test_download_requester_pays_item(
         )
         == 19554
     )
+
+
+async def test_download_custom_endpoint(
+    tmp_path: Path, requester_pays_asset_href: str, custom_endpoint_url: str
```

Comment:
This won't work, because there isn't a fixture named `custom_endpoint_url` ... e.g.:

```shell
$ pytest --network-access
-- >8 --
E       fixture 'custom_endpoint_url' not found
```

You'll either need to make a custom endpoint fixture, or just remove the tests if there isn't a good way to automatically test.

Reply:
You're right, I didn't notice that. I tested it locally, and for now, I'll remove the tests since I can't find an efficient way to automate them.